src/target/ext/host_io.rs

```diff
@@ -324,36 +320,32 @@ define_ext!(HostIoPwriteOps, HostIoPwrite);
 pub trait HostIoFstat: HostIo {
     /// Get information about the open file corresponding to fd.
     ///
-    /// On success the information is returned as a binary attachment and the
-    /// return value is the size of this attachment in bytes. If an error occurs
-    /// the return value is -1.
-    fn fstat(&mut self, fd: i32) -> HostIoResult<HostStat, Self>;
+    /// On success return a [`HostIoStat`] struct.
+    /// Return [`HostIoError::Errno`] if an error occurs.
+    fn fstat(&mut self, fd: u32) -> HostIoResult<HostIoStat, Self>;
 }
 
 define_ext!(HostIoFstatOps, HostIoFstat);
 
 /// Nested Target Extension - Host I/O unlink operation.
 pub trait HostIoUnlink: HostIo {
-    /// Delete the file at filename on the target.
-    ///
-    /// Return 0, or -1 if an error occurs.
+    /// Delete the file at filename on the target and return 0, or
+    /// [`HostIoError::Errno`] if an error occurs.
     fn unlink(&mut self, filename: &[u8]) -> HostIoResult<u32, Self>;
```

Comment:
see my comment about setfs wrt "return 0 on success". the same thing applies to unlink